drivers/gnss/gnss_u_blox_m10.c

```diff
@@ -484,7 +462,7 @@ static int ubx_m10_configure_gnss_device_baudrate(const struct device *dev)
 
 static int ubx_m10_configure_messages(const struct device *dev)
 {
-	int ret;
+	int ret = 0;
```

Comment:
```suggestion
	int ret;
```
not needed/consistent

Reply:
I got a compiler warning that's why I initialized ret.

Reply:
yeah this has to stay if the function ends with `return ret;` and you need that for the unlock

Reply:
> I got a compiler warning that's why I initialized ret.

Weird as `ret` gets assigned [here](https://github.com/zephyrproject-rtos/zephyr/pull/75845/files/b96993bf92d53bec200c437d8b399529cc474bd3#diff-36d7afa23d79157f63556261f9b5a9c322d43ada7bcad8deaadd42a8912bbb96R480) and it will happen at least once or am I missing something?

Reply:
yeah, below as well but I guess the compiler is not smart enough to detect that cycle will always be executed? this is really beyond me

Reply:
I'd bet on `buggy enough` rather than `not smart enough`. :thinking:

Reply:
It may be a feature 👀

Reply:
by the way the loop below should be

`for (int i = 0; i < ARRAY_SIZE(message_enable); ++i) {`

right now it happens to match because the array is 1 byte elements but it would not work in any other case

Reply:
Tried to play a bit with this on godbolt but I can't really reproduce the warning: https://godbolt.org/z/7Kcx16hdc, weird.

Reply:
@tomi-font sorry for obsessing on this, what environment are you using? I tried this locally

```diff
index 9d67fdc9dcc..6c8689ea532 100644
--- a/drivers/gnss/gnss_u_blox_m10.c
+++ b/drivers/gnss/gnss_u_blox_m10.c
@@ -462,7 +462,7 @@ static int ubx_m10_configure_gnss_device_baudrate(const struct device *dev)
 
 static int ubx_m10_configure_messages(const struct device *dev)
 {
-       int ret = 0;
+       int ret;
        k_spinlock_key_t key;
        struct ubx_m10_data *data = dev->data;
        struct ubx_cfg_msg_payload payload;
```

`$ west build -p -b mimxrt1062_fmurt6 samples/drivers/gnss`

no warnings

Reply:
It's not me that reported that there is a warning here, it's @it-klinger. I didn't even try to reproduce anything locally. :smile:

Reply:
sorry, wrong tag :-) @it-klinger ^^^

Reply:
here we are:

/home/zephyr/zephyrproject/zephyr/drivers/gnss/gnss_u_blox_m10.c: In function 'ubx_m10_configure_messages':
/home/zephyr/zephyrproject/zephyr/drivers/gnss/gnss_u_blox_m10.c:515:16: warning: 'ret' may be used uninitialized [-Wmaybe-uninitialized]
  515 |         return ret;
      |                ^~~
/home/zephyr/zephyrproject/zephyr/drivers/gnss/gnss_u_blox_m10.c:465:13: note: 'ret' was declared here
  465 |         int ret;
      |             ^~~

Reply:
> by the way the loop below should be
> 
> `for (int i = 0; i < ARRAY_SIZE(message_enable); ++i) {`
> 
> right now it happens to match because the array is 1 byte elements but it would not work in any other case

Should I include this in my PR and force push?

Reply:
> Should I include this in my PR and force push?

up to you, this is fixing a bunch of stuff, may as well... it won't get merged before 3.7 is cut anyway

Reply:
> here we are:

And you don't have changes that are not pushed? Maybe try rebasing on `main`? What's your compiler/Zephyr SDK (don't know whether this one matters) versions?

Reply:
:thinking: Giving up on this.